### ISSUE TITLE
Add validation for unnamed ops

### ISSUE THREAD

**Issue by hinerm** [scijava/scijava#162]
So, we have a weird situation where we have a [validate](https://github.com/scijava/incubator/blob/906bb6b3416c50478e982765a0f23a5158351c4a/scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/util/Infos.java#L93) method that is used to check `OpInfos` after they're made... but we also have [OpImplData](https://github.com/scijava/incubator/blob/906bb6b3416c50478e982765a0f23a5158351c4a/scijava/scijava-ops-indexer/src/main/java/org/scijava/ops/indexer/OpImplData.java#L150) that is doing parsing but **not** doing any validation.

Why do we check e.g. single outputs in `Infos` instead of when parsing the `OpImplData`?

I want to add a check for unnamed Ops. It seems like `OpImplData` would be a reasonable place for this because, [as indicated](https://github.com/scijava/incubator/blob/f4ce2135f01e2850f884b551de5035fef5f9dc8c/scijava/scijava-ops-indexer/src/main/java/org/scijava/ops/indexer/OpImplData.java#L61-L62), there must be at least one name - but this is never checked anywhere.

Basically, if we want to do runtime checking for Op validity, we use the system as-is. If we want to do compile-time checking instead, we would migrate the [InvalidOpExceptions](https://github.com/scijava/incubator/blob/906bb6b3416c50478e982765a0f23a5158351c4a/scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/exceptions/InvalidOpException.java#L42) to `scijava-ops-indexer`

Either way we can then add a check for Unnamed ops

**Comment by hinerm** [comment]
@gselzer @ctrueden I am partial for compile-time validation instead of runtime. How do you feel?
